File: php/features.php
<?php
declare(strict_types=1);

// BuddhaQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BuddhaQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BuddhaQuotesBaseFeature();
            case "test":
                return new BuddhaQuotesTestFeature();
            default:
                return new BuddhaQuotesBaseFeature();
        }
    }
}

File: py/buddhaquotes_sdk/features.py
# BuddhaQuotes SDK feature factory

from buddhaquotes_sdk.feature.base_feature import BuddhaQuotesBaseFeature
from buddhaquotes_sdk.feature.test_feature import BuddhaQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BuddhaQuotesBaseFeature(),
        "test": lambda: BuddhaQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
